metal/devices.go

```diff
@@ -188,6 +193,66 @@ func (i *instances) InstanceShutdownByProviderID(_ context.Context, providerID s
 	return device.State == "inactive", nil
 }
 
+// InstanceShutdown returns true if the node is shutdown in cloudprovider
+func (i *instances) InstanceShutdown(ctx context.Context, node *v1.Node) (bool, error) {
+	return i.InstanceShutdownByProviderID(ctx, node.Spec.ProviderID)
+}
+
+// InstanceExists returns true if the node exists in cloudprovider
+func (i *instances) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
+	return i.InstanceExistsByProviderID(ctx, node.Spec.ProviderID)
+}
+
+// InstanceMetadata returns instancemetadata for the node according to the cloudprovider
+func (i *instances) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprovider.InstanceMetadata, error) {
+	device, err := i.deviceByNode(node)
+	if err != nil {
+		return nil, err
+	}
+	nodeAddresses, err := nodeAddresses(device)
+	if err != nil {
+		// TODO(displague) we error on missing private and public ip. is that restrictive?
+
+		// TODO(displague) should we return the public addresses DNS name as the Type=Hostname NodeAddress type too?
+		return nil, err
+	}
+	var p, r, z string
+	if device.Plan != nil {
+		p = device.Plan.Slug
+	}
+
+	// "A zone represents a logical failure domain"
+	// "A region represents a larger domain, made up of one or more zones"
+	//
+	// Equinix Metal metros are made up of one or more facilities, so we treat
+	// metros as K8s topology regions. EM facilities are then equated to zones.
+	//
+	// https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone
+
+	if device.Facility != nil {
+		z = device.Facility.Code
+	}
+	if device.Metro != nil {
+		r = device.Metro.Code
+	}
+
+	return &cloudprovider.InstanceMetadata{
+		ProviderID:    node.Spec.ProviderID,
```

Comment:
Aha, here! This is where you are expected to set the provider ID, but it is just passing back what it received.

I believe this should be `fmt.Sprintf("%s://%s", ProviderName, device.ID)`

Reply:
Confirmed that the above works.

Reply:
@deitch I just want to say, VSCode Copilot is amazing.. 

![image](https://user-images.githubusercontent.com/317653/144240957-26143870-3893-44b3-9034-768e3fbc198e.png)

It's like a portable @deitch.

Reply:
haha! Only as portable as your laptop. 😁 

Make the changes, I can approve and merge this in. Already fully tested.